packages/@lwc/module-resolver/src/index.ts

```diff
@@ -6,154 +6,109 @@
  */
 /* eslint-env node */
 
-import glob from 'fast-glob';
 import path from 'path';
 import fs from 'fs';
-import nodeModulePaths from './node-modules-paths';
-
-const DEFAULT_IGNORE = ['**/node_modules/**', '**/__tests__/**'];
-const PACKAGE_PATTERN = ['*/*/package.json', '*/package.json', 'package.json'];
-const MODULE_ENTRY_PATTERN = `**/*.([jt]s|html|css)`;
-const LWC_CONFIG_FILE = '.lwcrc';
+import {
+    getModuleEntry,
+    normalizeConfig,
+    loadConfig,
+    mergeModules,
+    LWC_CONFIG_FILE,
+} from './utils';
 
 export interface RegistryEntry {
     entry: string;
-    moduleSpecifier: string;
-    moduleName: string;
-    moduleNamespace: string;
+    specifier: string;
 }
 
-export interface ModuleResolverConfig {
-    moduleDirectories: string[];
-    rootDir: string;
-    modulePaths: string[];
-    ignorePatterns?: string[];
+export interface ModuleRegistryMap {
+    [key: string]: RegistryEntry;
 }
 
-interface FlatEntry {
-    path: string;
+export interface ModuleResolverConfig {
+    rootDir: string;
+    modules: ModuleId[];
 }
 
-function createRegistryEntry(entry, moduleSpecifier, moduleName, moduleNamespace): RegistryEntry {
-    return { entry, moduleSpecifier, moduleName, moduleNamespace };
+export type ModuleId = string | [string, string];
```

Comment:
i would prefer splitting these into 2 distinct types to improve maintenance and readability

Reply:
Fair enough